src/main/database/adapters/base.ts

```diff
@@ -0,0 +1,39 @@
+export abstract class DatabaseAdapter<DatabaseItemType> {
```

Comment:
This is a base class to implement database adapters, is an initial concept but the idea behind this is that you can plug in whatever underlying database you want, even an in memory one, or a remote one

Reply:
If the class is abstract and all the methods are abstract as well why don't use an interface?

Reply:
because interfaces doesn't allow you to add base methods like in an abstract class that you will need to implement otherwise, imagine we need to add a common method to all the classes that uses this interface, we would need to manually add it to each class, with an abstract class we can simply add the common method to the abstract class

Reply:
Hmm, but we are assuming all extensions of this class will need that and they might not. If the goal is to be able to "plug in whatever underlying database you want" we need to have only the methods needed to interact with the database. In this regard, the method `connect` should not be part of it either as not all databases might require to connect. There is an example on the test, you need to declare a method that is not being used anywhere since the adapter is memory and it does not need to connect.

Said that I don't really mind keeping it as an abstract class but then the classes 
`DriveFilesDB` and `DriveFoldersDB` should extend and not implement. By implementing the class any method even the non-abstract needs to be implemented.

Reply:
I don't get the point of extending an abstract class instead of implementing it, it was on purpose, all the adapters need to implement the methods described in the abstract class (create, get, update and remove), if you don't need `connect` for your adapter case, just return a promise like this `Promise.resolve({success: true})` and you are done, but the other methods are mandatory for the adapter to work. There's always cases where a class doesn't fit, but I believe this implementation is flexible enough to solve those cases, if you create an inMemoryAdapter, just return a promise as I said and implement the other methods as the abstract class defines.

The test case doesn't fail because all the methods are mocked using `jest.fn` but if you check the DriveFilesDB, when you try to do an operation without calling connect first, it throws an error, that's on purpose because you need to open a connection first